File: diagonal-difference/diagonal-difference.go
package main

import (
	"math"
)

func diagonalDifference(arr [][]int32) int32 {
	primaries := diagonals(arr)
	primariesSum := sum(primaries)

	var arrReversed [][]int32
	for i := len(arr) - 1; i >= 0; i-- {
		arrReversed = append(arrReversed, arr[i])
	}
	secondaries := diagonals(arrReversed)
	secondariesSum := sum(secondaries)

	return int32(math.Abs(float64(primariesSum) - float64(secondariesSum)))
}

func diagonals(arr [][]int32) []int32 {
	var result []int32
	for i, ar := range arr {
		result = append(result, ar[i])
	}
	return result
}

func sum(arr []int32) int32 {
	var result int32 = 0
	for _, n := range arr {
		result += n
	}
	return result
}

File: diagonal-difference/DiagonalDifference.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class DiagonalDifference {
  
  public static int diagonalDifference(List<List<Integer>> matrix) {
    List<Integer> primaries = getDiagonals(matrix);
    int primariesSum = getSum(primaries);

    Collections.reverse(matrix);
    List<Integer> secondaries = getDiagonals(matrix);
    int secondariesSum = getSum(secondaries);

    return (int) Math.abs(
      (double) primariesSum - (double) secondariesSum
    );
  }

  private static List<Integer> getDiagonals(List<List<Integer>> matrix) {
    List<Integer> result = new ArrayList<>();

    for (int i = 0; i < matrix.size(); i++) {
      List<Integer> ns = matrix.get(i);
      int n = ns.get(i);
      result.add(n);
    }

    return result;
  }

  private static int getSum(List<Integer> ns) {
    return ns
      .stream()
      .reduce(0, Integer::sum);
  }
}
